api/src/test/java/org/openmrs/module/locationbasedaccess/aop/UserSearchAdviserTest.java

```diff
@@ -0,0 +1,472 @@
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.openmrs.Person;
+import org.openmrs.User;
+import org.openmrs.Role;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.aop.common.AOPContextSensitiveTest;
+import org.openmrs.module.locationbasedaccess.aop.common.TestWithAOP;
+import org.openmrs.module.locationbasedaccess.aop.interceptor.UserServiceInterceptorAdvice;
+
+public class UserSearchAdviserTest extends AOPContextSensitiveTest {
+    
+    private UserService userService;
+    private PersonService personService;
+    private LocationService locationService;
+    private static final String XML_FILENAME_WITH_ADMIN_USER_DATA = "include/AdminUserData.xml";
+    private static final String XML_FILENAME_WITH_PERSON_DATA = "include/PatientTestData.xml";
+    private static final String DEMO_USER_PROPERTY= "locationUuid";
+    
+    private static final int DEMO_LOCATION_ID = 1;
+    private static final int DEMO_USER1_ID = 6001;
+    private static final int DEMO_USER2_ID = 6002;
+    private static final int DEMO_PERSON1_ID = 6001;
+    private static final int DEMO_PERSON2_ID = 6002;
+ 
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(UserServiceInterceptorAdvice.class);
+        testCase.addService(UserService.class);
+    }
+    
+    public UserSearchAdviserTest() {
+    }
+
+    @Before
+    public void setUp()  throws Exception {
+        userService = Context.getUserService();
+        personService = Context.getPersonService();
+        locationService = Context.getLocationService();
+        executeDataSet(XML_FILENAME_WITH_ADMIN_USER_DATA);
+        executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+    }
+        
+    @Test
+    public void getUsers_adminCanAccessUsersFromAllLocations(){
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location = locationService.getLocation(DEMO_LOCATION_ID);
+
+        List<Role> userRole = new ArrayList<Role>();
+        userRole.add(userService.getRole("userRole")); //Adding the role "userRole" to a new List
+        List<User> userList = userService.getUsers("User",userRole,true);
+        assertEquals(userList.size(), 2);
+        assertNotNull(userList.get(0));
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),location.getUuid());
+        assertNotNull(userList.get(1));
+        assertEquals(userList.get(1).getUserProperty(DEMO_USER_PROPERTY),"");
+    }
+    
+    @Test
+    public void getUsers_nonAdminUserWithOutLocationPropertyShouldNotGetUserInformation(){

```

Comment:
By this test method, did you tested that the user still can't access others user objects without required location user properties?

Reply:
In the most recent commit I did @suthagar23